library/ssl_tls13_client.c

```diff
@@ -1015,22 +1021,43 @@ int mbedtls_ssl_write_pre_shared_key_ext( mbedtls_ssl_context *ssl,
 
     for ( int i = 0; ciphersuites[i] != 0; i++ )
     {
-        suite_info = mbedtls_ssl_ciphersuite_from_id( ciphersuites[i] );
+        /* In this implementation we only add one pre-shared-key extension.
+         * If we are using resumption then we use this ciphersuite. 
+         */
+        if( ssl->handshake->resume == 1 )
+        {
+            suite_info = mbedtls_ssl_ciphersuite_from_id( ssl->session_negotiate->ciphersuite );
 
-        if( suite_info == NULL )
-            continue;
+            if( suite_info == NULL )
+                continue;
 
-        hash_len = mbedtls_hash_size_for_ciphersuite( suite_info );
+            hash_len = mbedtls_hash_size_for_ciphersuite( suite_info );
 
-        if( hash_len == -1 )
-        {
-            MBEDTLS_SSL_DEBUG_MSG( 1, ( "mbedtls_hash_size_for_ciphersuite == -1, mbedtls_ssl_write_pre_shared_key_ext failed" ) );
-            return( MBEDTLS_ERR_SSL_INTERNAL_ERROR );
+            if( hash_len == 0 )
+            {
+                MBEDTLS_SSL_DEBUG_MSG( 1, ( "mbedtls_hash_size_for_ciphersuite == 0, mbedtls_ssl_write_pre_shared_key_ext failed" ) );
+                return( MBEDTLS_ERR_SSL_INTERNAL_ERROR );
+            }
+            ssl->handshake->ciphersuite_info = suite_info;
         }
```

Comment:
This block (`resume == 1`) could be moved out of the `for` loop as it is not use anything from `ciphersuites`.